### DISCUSSION TITLE
new board based on stm32f405 with sd configured in spi not sdio

### DISCUSSION THREAD

**Discussion by hakimoune** [micropython/micropython#13304] [Hardware & Peripherals]
Hello,
how to configure a card with same config as pyboard with stm32f405 but the sd card is connected with spi instead of sdio.
thank you

**Comment by DavesCodeMusings** [comment]
Below is a working example of an SPI attached microSD card. (I'm using ESP32, not pyboard, so you'll need to adjust the slot and pin numbers.)

```
from machine import Pin, SDCard
from os import VfsFat, mount

SD_MOUNT = "/pub"
print(f"Mounting micro SD card on: {SD_MOUNT}")
sdcard = SDCard(slot=3, sck=Pin(12), miso=Pin(13), mosi=Pin(11), cs=Pin(10))
vfs = VfsFat(sdcard)
mount(vfs, SD_MOUNT)
```

**Reply by hakimoune** [reply]
thank you for your response, this is in runtime but how do we do it before porting (Build and Flash) ?

**Reply by DavesCodeMusings** [reply]
Sorry, I misunderstood your original question. I have no idea.
